ui/core/src/model/query.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Definition, UnknownSpec } from './definitions';
+
+export type QueryType = 'TimeSeriesQuery';
+
+interface QuerySpec<PluginSpec> {
+  plugin: Definition<PluginSpec>;
+}
+
+export interface QueryDefinition<Kind = QueryType, PluginSpec = UnknownSpec> {
```

Comment:
A generic query definition interface that can extend to support more than just `TimeSeriesQuery`